src/Device/includes/Sampling.h

```diff
@@ -315,6 +316,40 @@ HIPRT_HOST_DEVICE HIPRT_INLINE float3 GGX_VNDF_spherical_caps_sample(const float
     return hippt::normalize(make_float3(alpha_x * Nh.x, alpha_y * Nh.y, Nh.z));
 }
 
+/**
+ * [Bounded VNDF Sampling for Smith–GGX Reflections, Kenta Eto and Yusuke Tokuyoshi; 2023]
+ */
+HIPRT_HOST_DEVICE HIPRT_INLINE float3 GGX_VNDF_bounded_sample(const float3 local_view_direction, float alpha_x, float alpha_y, Xorshift32Generator& random_number_generator)
```

Comment:
The overall implementation of the sampling function looks quite different from the reference which makes it harder to understand and follow.

I suppose this is all in the name of performance but in the grand scheme of things, simplifying a few float ops in a sampling function isn't worth it compared to the introduced divergence of the code from the reference in the paper.

I'd prefer an implementation closer to the one presented in the paper.